### PR TITLE
add python solution to triangles 1015

### PR DESCRIPTION
**Pull request by jjliewie**
_If any of the below don't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which include the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making the changes.

### CODE REVIEW

File: solutions/usaco-1015.mdx [hunk 1]
```diff
@@ -226,4 +226,62 @@ public class Triangles {
...
+
+```py
+import sys
+
+sys.stdin = open('triangles.in', 'r')
+sys.stdout = open('triangles.out', 'w')
+
+t = sys.stdin.readline
+
+def solution():
```
Comment (SansPapyrus683) [inline solutions/usaco-1015.mdx:239]: why have this function

Comment (jjliewie) [reply solutions/usaco-1015.mdx:239]: bro u are like my ib lit teacher.

Comment (jjliewie) [reply solutions/usaco-1015.mdx:239]: but its bc i like having a solution function

Comment (SansPapyrus683) [reply solutions/usaco-1015.mdx:239]: what

Comment (jjliewie) [reply solutions/usaco-1015.mdx:239]: its not a bad thing my ib lit teacher is pretty cool.

Comment (jjliewie) [reply solutions/usaco-1015.mdx:239]: >:(

File: solutions/usaco-1015.mdx [hunk 2]
```diff
@@ -226,4 +226,59 @@ public class Triangles {
...
+```py
+import sys
+
+sys.stdin = open('triangles.in', 'r')
+sys.stdout = open('triangles.out', 'w')
+
+t = sys.stdin.readline
+
+n = int(t())
+fieldx, fieldy = [[] for _ in range(20001)], [[] for _ in range(20001)]
```
Comment (SansPapyrus683) [inline solutions/usaco-1015.mdx:240]: magic number

File: solutions/usaco-1015.mdx [hunk 3]
```diff
@@ -226,4 +226,59 @@ public class Triangles {
...
 
+<PySection>
+
+```py
+import sys
+
+sys.stdin = open('triangles.in', 'r')
+sys.stdout = open('triangles.out', 'w')
+
+t = sys.stdin.readline
```
Comment (SansPapyrus683) [inline solutions/usaco-1015.mdx:237]: use `with`, you know the drill

File: solutions/usaco-1015.mdx [hunk 4]
```diff
@@ -226,4 +226,59 @@ public class Triangles {
...
+			psumy[i][fieldy[i][j+1]] = temp
+    
+ans = 0
+for a in allcoors:
+	x, y = a
+	if not psumx[x] or not psumy[y]:
+		continue
+	ans += abs(psumx[x][y] * psumy[y][x]) 
+
+print(ans % (int(1e9+7)))
```
Comment (SansPapyrus683) [inline solutions/usaco-1015.mdx:279]: spaces between operators

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/usaco-1015.mdx (hunk 3 above)
Comment (jjliewie) [reply solutions/usaco-1015.mdx]: done

File: solutions/usaco-1015.mdx (hunk 2 above)
Comment (jjliewie) [reply solutions/usaco-1015.mdx]: done

**Review by jjliewie — COMMENTED** [review]
addressed everything

File: solutions/usaco-1015.mdx [hunk 5]
```diff
@@ -226,4 +226,56 @@ public class Triangles {
 
 </JavaSection>
 
+<PySection>
+
+```py
+MAXIMUM = 10000
+MOD = int(1e9 + 7)
+
+fieldx, fieldy = [[] for _ in range((MAXIMUM * 2) + 1)], [[] for _ in range((MAXIMUM * 2) + 1)]
```
Comment (SansPapyrus683) [inline solutions/usaco-1015.mdx:235]: separate lines

File: solutions/usaco-1015.mdx [hunk 6]
```diff
@@ -226,4 +226,56 @@ public class Triangles {
 
 </JavaSection>
 
+<PySection>
+
+```py
+MAXIMUM = 10000
```
Comment (SansPapyrus683) [inline solutions/usaco-1015.mdx:232]: i'd rename this to just MAX

Comment (jjliewie) [reply solutions/usaco-1015.mdx:232]: I would rather not. can i please not i like maximum.

Comment (SansPapyrus683) [reply solutions/usaco-1015.mdx:232]: how, it's literally the same thing

Comment (jjliewie) [reply solutions/usaco-1015.mdx:232]: pls let me keep it im begging you

File: solutions/usaco-1015.mdx (hunk 5 above)
Comment (jjliewie) [reply solutions/usaco-1015.mdx:235]: done

File: solutions/usaco-1015.mdx (hunk 6 above)
Comment (SansPapyrus683) [reply solutions/usaco-1015.mdx:232]: what do you have against changing it??

Comment (jjliewie) [reply solutions/usaco-1015.mdx:232]: everything let me keep it.

File: solutions/usaco-1015.mdx [hunk 7]
```diff
@@ -225,5 +224,59 @@ public class Triangles {
...
+	y.sort()
+
+psumx = [[] for _ in range((MAX * 2) + 1)]
+psumy = [[] for _ in range((MAX * 2) + 1)]
+	
+for i in range((MAX * 2) + 1):
+	xl, yl = len(fieldx[i]), len(fieldy[i])
+	psumx[i], psumy[i] = {}, {}
+	if xl > 1:
+		temp = sum(fieldx[i]) - xl * fieldx[i][0]
```
Comment (SansPapyrus683) [inline solutions/usaco-1015.mdx:257]: bad name

File: solutions/usaco-1015.mdx [hunk 8]
```diff
@@ -225,5 +224,59 @@ public class Triangles {
...
+		for j in range(xl - 1):
+			temp += (2 * (j + 1) - xl) * (fieldx[i][j + 1] - fieldx[i][j])
+			psumx[i][fieldx[i][j + 1]] = temp
+
+	if yl > 1:
+		temp = sum(fieldy[i]) - yl * fieldy[i][0]
+		psumy[i][fieldy[i][0]] = temp
+		for j in range(yl - 1):
+			temp += (2 * (j + 1) - yl) * (fieldy[i][j + 1] - fieldy[i][j])
+			psumy[i][fieldy[i][j + 1]] = temp
```
Comment (SansPapyrus683) [inline solutions/usaco-1015.mdx:268]: i'm begging you add some comments

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/usaco-1015.mdx [hunk 9]
```diff
@@ -225,5 +224,59 @@ public class Triangles {
...
+	x.sort()
+for y in fieldy:
+	y.sort()
+
+psumx = [[] for _ in range((MAX * 2) + 1)]
+psumy = [[] for _ in range((MAX * 2) + 1)]
+	
+for i in range((MAX * 2) + 1):
+	xl, yl = len(fieldx[i]), len(fieldy[i])
+	psumx[i], psumy[i] = {}, {}
```
[the remark this answers is not in the document]
Comment (SansPapyrus683) [reply solutions/usaco-1015.mdx:255]: then why not define them as sets above in the listcomp?

File: solutions/usaco-1015.mdx (hunk 8 above)
Comment (SansPapyrus683) [reply solutions/usaco-1015.mdx:268]: alr i'll add some later then

**Pull request closed without merging**